Sources/ForageSDK/Foundation/Network/LDManager.swift

```diff
@@ -11,12 +11,79 @@ import LaunchDarkly
 
 internal protocol LDClientProtocol {
     func doubleVariationWrapper(forKey key: String, defaultValue: Double) -> Double
+    func jsonVariationWrapper(forKey key: String, defaultValue: LDValue) -> LDValue
 }
 
 extension LDClient: LDClientProtocol {
     func doubleVariationWrapper(forKey key: String, defaultValue: Double) -> Double {
         return self.doubleVariation(forKey: key, defaultValue: defaultValue)
     }
+    
+    func jsonVariationWrapper(forKey key: String, defaultValue: LDValue) -> LDValue {
+        return self.jsonVariation(forKey: key, defaultValue: defaultValue)
+    }
+}
+
+extension LDValue {
+    
+    /// Converts the original LDValue returned from the polling feature flag to an Array of Ints.
+    internal func convertToIntArray() -> [Int] {
+        let intervalArrayAsLDValue = self.extractArray(pollingFlagObj: self)
+        let intervalArrayAsArray = self.convertToArray(pollingArray: intervalArrayAsLDValue)
+        return self.convertArrayValsToInts(arrayOfInts: intervalArrayAsArray)
+    }
+    
+    /// Inspects the original LDValue returned from the polling feature flag and extracts the value mapped to INTERVALS
+    ///
+    /// - Parameters:
+    ///   - pollingFlagObj: The `LDValue` representing a JSON object.
+    internal func extractArray(pollingFlagObj: LDValue) -> LDValue {
+        let defaultLDValueArray: LDValue = LDValue(arrayLiteral: [])
+        switch pollingFlagObj {
+        case .object(let dictValue):
+            for (key, value) in dictValue {
+                if (key == INTERVALS) {
+                    return value
+                }
+            }
+            return defaultLDValueArray
+        default:
+            return defaultLDValueArray
+        }
+    }
+    
+    /// Converts the LDValue to an Array.
+    ///
+    /// - Parameters:
+    ///   - pollingArray: The `LDValue` representing an Array.
+    internal func convertToArray(pollingArray: LDValue) -> [LDValue] {
+        switch pollingArray {
+        case .array(let intervalVals):
+            return intervalVals
+        default:
+            return []
+        }
+    }
+
+    /// Converts an Array of LDValue to an Array of Ints.
+    ///
+    /// - Parameters:
+    ///   - arrayOfInts: The `LDValue` representing an Array of Ints.
+    internal func convertArrayValsToInts(arrayOfInts: [LDValue]) -> [Int] {
+        return arrayOfInts.map { timeInMs in
+            switch timeInMs {
+            case .number(let valAsNum):
+                return Int(valAsNum)
+            default:
+                return 1000
+            }
+        }
+    }
+}
+
+internal protocol LDManagerProtocol {
+    func getVaultType(ldClient: LDClientProtocol?, genRandomDouble: () -> Double, fromCache: Bool) -> VaultType
+    func getPollingIntervals(ldClient: LDClientProtocol?) -> [Int]
```

Comment:
can we mark `ldClient` as required if it's meant to be set by the caller?

@dleis612

Reply:
Uhhhh you're probably right!

Reply:
discussed via slack! We can leave it as-is